DataDonationCryptoObjC/DataDonationCryptoObjC/Sources/DataDonationCryptoObjC.swift

```diff
@@ -16,9 +16,22 @@
 
 import Foundation
 
-@objc public protocol KeychainKeyProviderProtocol {
-    @objc func getDonorPrivateKey(for programName: String) throws -> String
-    @objc func getDonorPublicKey(for programName: String) throws -> String
-    @objc func removeDonorKeyPair(for programName: String) throws
-    @objc func storeDonorKeyPairData(_ data: Data, for programName: String) throws
+@objc public protocol DataDonationCryptorObjCProtocol {
+    @objc func encrypt(_ plainBody: Data, base64EncodedPublicKey: String) throws -> Data 
+    @objc func decrypt(_ encryptedData: Data, base64EncodedPrivateKey: String) throws -> Data
 }
+
+@objc public protocol DataDonationSignerObjCProtocol {
+    @objc func sign(data: Data, isSalted: Bool, donorKeyIdentifier: String) throws -> Data
+    @objc func verify(data: Data, signature: Data, isSalted: Bool, donorKeyIdentifier: String) throws
+}
+
+@objc public protocol DataDonationKeychainStoreObjCProtocol {
+    @objc func generateDonorKeyPair(with keyIdentifier: String) throws
+    @objc func storeDonorKeyPairData(_ data: Data, with keyIdentifier: String) throws
+    @objc func fetchDonorPrivateKeyAsBase64(with keyIdentifier: String) throws -> String
+    @objc func fetchDonorPublicKeyAsBase64(with keyIdentifier: String) throws -> String
+    @objc func deleteDonorKeyPair(with keyIdentifier: String) throws
```

Comment:
Currently we would need to call always delete and then store, if a key for a program is present, unless we have a method to look if a given KeyPair is the same as in the storage. Which mean either we have such a method on the platform or as mentioned we delete and save always.